node/flatpak_node_generator/providers/npm.py

```diff
@@ -94,12 +94,74 @@ def _process_packages_v1(
             if 'dependencies' in info:
                 yield from self._process_packages_v1(lockfile, info)
 
+    def _process_packages_v2(
+        self, lockfile: Path, entry: Dict[str, Dict[Any, Any]]
+    ) -> Iterator[Package]:
+        for install_path, info in entry.get('packages', {}).items():
+            if (info.get('dev') or info.get('devOptional')) and self.no_devel:
+                continue
+            if info.get('link'):
+                # NOTE We're not interested in symlinks, NPM will create them at install time
+                # but we still could collect package symlinks anyway just for completeness
+                continue
+
+            name = info.get('name')
+
+            source: PackageSource
+            package_json_path = lockfile.parent / install_path / 'package.json'
+            if package_json_path.exists():
```

Comment:
Is there any chance this will trigger if someone has already run `npm install`, so a registry package would be in `node_modules`?

Reply:
It will. But running the generator against a dirty source tree will screw up the generated sources anyway, so I'm not sure if it makes sense to add a safety check just here.

Reply:
oh that's unfortunate :sweat_smile: since when is that the case? I think it used to work at *some* point...

Not really an issue for this PR regardless, though.

Reply:
IIRC it always was the case. I mean, if a `package-lock.json` or `yarn.lock` file somehow gets into a release tarball in the registry, we'll be generating unused dependencies from there as well, potentially exploding the size of `generated-sources.json`.

Reply:
In any case, added a condition check assuming that a local source can't be inside a `node_modules` directory.